### PR TITLE
Add security headers to site

### PR DESCRIPTION
**Pull request by kagarmoe**
Signed-off-by: kagarmoe <kgarmoe@chef.io>

## Description

Improves site security.

## Issues Resolved

https://securityheaders.com/?q=docs.chef.io&hide=on&followRedirects=on 

## Check List

- [ ] Spell Check
- [ ] Local build
- [ ] Examine the local build
- [ ] All tests pass

### CODE REVIEW

**Review by tas50 — APPROVED** [review]

File: netlify.toml [hunk 1]
```diff
@@ -18,6 +18,15 @@
...
   command = "make assets; hugo --gc --minify --enableGitInfo --buildFuture"
 
+[[headers]]
+  for = "/*"
+  [headers.values]
+    X-Frame-Options = "DENY"
+    X-Content-Type-Options = "nosniff"
+    Referrer-Policy = "no-referrer-when-downgrade"
+    Strict-Transport-Security = "max-age=2592000"
+    Permissions-Policy = "vibrate=(), geolocation=(), midi=(), notifications=(), push=(), sync-xhr=(), microphone=(), camera=(), magnetometer=(), gyroscope=(), speaker=(), vibrate=(), fullscreen=(), payment=()"
```
Comment (mimaslanka) [inline netlify.toml:29]: From a security perspective, this looks reasonable to me. You should also have Content-Security-Policy and X-XSS-Protection set as well.

Comment (kagarmoe) [reply netlify.toml:29]: I spent two hours trying to work out the `Content-Security-Policy`. We have a lot of arbitrary JS. And a lot of external sources.  :-(

Comment (kagarmoe) [reply netlify.toml:29]: I made an issue for it:https://github.com/chef/chef-web-docs/issues/3501

Comment (mimaslanka) [reply netlify.toml:29]: Looks good to me

**Review by mimaslanka — APPROVED** [review]

**Pull request merged**